metadata-ingestion/src/datahub/ingestion/source/tableau.py

```diff
@@ -1085,14 +1320,18 @@ def emit_datasource(
             aspects=[self.get_data_platform_instance()],
         )
 
-        datasource_name = datasource.get("name") or datasource_id
-        if is_embedded_ds and workbook and workbook.get("name"):
-            datasource_name = f"{workbook['name']}/{datasource_name}"
         # Browse path
-        browse_paths = BrowsePathsClass(
-            paths=[f"/{self.config.env.lower()}/{self.platform}/{project}"]
-        )
-        dataset_snapshot.aspects.append(browse_paths)
+
+        if browse_path and is_embedded_ds and workbook and workbook.get("name"):
```

Comment:
this is confuing. why do we need all these conditions? why does it matter if it's an embedded datasource vs not embedded? shouldn't we always just try to build a browse path???